File: app/src/main/java/ru/vincetti/vimoney/data/models/TransactionListModel.java
package ru.vincetti.vimoney.data.models;

import androidx.room.ColumnInfo;

import java.util.Date;

public class TransactionListModel {

    public int id;
    @ColumnInfo(name = "account_name")
    public String accountName;
    public String description;
    public Date date;
    public int type;
    public float sum;
    @ColumnInfo(name = "account_symbol")
    public String curSymbol;

    public int getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public Date getDate() {
        return date;
    }

    public int getType() {
        return type;
    }

    public float getSum() {
        return sum;
    }

    public String getAccountName() {
        return accountName;
    }

    public String getCurSymbol() {
        return curSymbol;
    }
}

File: app/src/main/java/ru/vincetti/vimoney/data/models/TransactionListModel.kt
package ru.vincetti.vimoney.data.models

import androidx.room.ColumnInfo
import java.util.*

data class TransactionListModel(
    val id: Int = 0,

    @ColumnInfo(name = "account_name")
    val accountName: String,

    val description: String,

    @ColumnInfo(name = "category_icon")
    val symbol: String,

    val date: Date,

    val type: Int,

    val sum: Float,

    @ColumnInfo(name = "account_symbol")
    val curSymbol: String,

    @ColumnInfo(name = "extra_key")
    val extraKey: String,

    @ColumnInfo(name = "extra_value")
    val extraValue: String
) {

    fun getTypeString(): String {
        return when (type) {
            TransactionModel.TRANSACTION_TYPE_INCOME -> "+"
            TransactionModel.TRANSACTION_TYPE_TRANSFER -> "="
            TransactionModel.TRANSACTION_TYPE_DEBT -> return "|"
            else -> "-"
        }
    }
}
